doc/transceiver-monitor-hld.md

```diff
@@ -0,0 +1,140 @@
+	rx1power                = FLOAT                                      ; rx1 power in dbm
+	rx2power                = FLOAT                                      ; rx2 power in dbm
+	rx3power                = FLOAT                                      ; rx3 power in dbm
+	rx4power                = FLOAT                                      ; rx4 power in dbm
+	tx1bias                 = FLOAT                                      ; tx1 bias in mA
+	tx2bias                 = FLOAT                                      ; tx2 bias in mA
+	tx3bias                 = FLOAT                                      ; tx3 bias in mA
+	tx4bias                 = FLOAT                                      ; tx4 bias in mA
+
+
+### 1.2 Access eeprom from platform container ###
+
+Transceiver information eeprom can be accessed via read files(e.g. /sys/bus/i2c/devices/2-0048/hwmon/hwmon4/qsfp9_eeprom), different vendors may have these files under different folders, these folder need to be mounted to platform container so Xcvrd can access them. 
+
+Another potential enhancement to the eeprom reading is to only read the needed parameters out instead of read all of them, this can be achieved by add new API to `SfpUtilBase` which can read desired bytes starting from give offset and parse the bytes to readable format accordingly.
+
+For the convenience of implementation and reduce the time consuming, need to enhance the `SfpUtilBase` class to provide functions to get `eeprom_if_dict` and `eeprom_dom_dict` separately, the intrested values are defined  in section 1.1.1 and 1.1.2, we can pick up these values from eeprom by calling the above new API with proper offset and number of bytes. 
+
+
+### 1.3 Transceiver plug in/out event ###
+
+Xcvrd need to be triggered by transceiver plug in/out event to refresh the transceiver info table.
+
+Transceiver plug in/out status can be derived from the content of sysfs file like `"/sys/bus/i2c/devices/2-0048/hwmon/hwmon7/qsfp10_status"`, if the content of the file is "good" represent SFP is present, conent change to "not_connected" means SFP plug out.
+
+To monitor the file change, can introduce python lib [inotify](https://pypi.org/project/inotify/), which can raise notification when target file change. Below is a sample for how to use inotify lib to monitor file change:
+
+    i = inotify.adapters.Inotify()
+
+    i.add_watch(b'/bsp/qsfp/qsfp10_status')
+
+    try:
+        for event in i.event_gen():
+            if event is not None:
+                (header, type_names, watch_path, filename) = event
+                _LOGGER.info("WD=(%d) MASK=(%d) COOKIE=(%d) LEN=(%d) MASK->NAMES=%s "
+                             "WATCH-PATH=[%s] FILENAME=[%s]",
+                             header.wd, header.mask, header.cookie, header.len, type_names,
+                             watch_path.decode('utf-8'), filename.decode('utf-8'))
+    finally:
+        i.remove_watch(b'/tmp')

```

Comment:
I understand this is just an example but looks like remove_watch does not match the add_watch
also /bsp/ is not available in pmon (it will run in pmon, right?) so it is better to use /sys/bus...

Reply:
yes, we should mount /sys/bus to pmon, will revise the example.